File: app/src/main/java/de/westnordost/streetcomplete/quests/bench_backrest/AddBenchBackrestForm.java
package de.westnordost.streetcomplete.quests.bench_backrest;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import de.westnordost.streetcomplete.R;
import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment;

public class AddBenchBackrestForm extends YesNoQuestAnswerFragment {
    public static final String PICNIC_TABLE = "picnic_table";

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
    {
        View view = super.onCreateView(inflater, container, savedInstanceState);
        addOtherAnswers();
        return view;
    }

    private void addOtherAnswers()
    {
        addOtherAnswer(R.string.quest_bench_answer_picnic_table, () -> {
            Bundle answer = new Bundle();
            answer.putBoolean(PICNIC_TABLE, true);
            applyAnswer(answer);
        });
    }
}

File: app/src/main/java/de/westnordost/streetcomplete/quests/bench_backrest/AddBenchBackrestForm.kt
package de.westnordost.streetcomplete.quests.bench_backrest

import de.westnordost.streetcomplete.R
import de.westnordost.streetcomplete.quests.AbstractOsmQuestForm
import de.westnordost.streetcomplete.quests.AnswerItem
import de.westnordost.streetcomplete.quests.bench_backrest.BenchBackrestAnswer.NO
import de.westnordost.streetcomplete.quests.bench_backrest.BenchBackrestAnswer.PICNIC_TABLE
import de.westnordost.streetcomplete.quests.bench_backrest.BenchBackrestAnswer.YES

class AddBenchBackrestForm : AbstractOsmQuestForm<BenchBackrestAnswer>() {

    override val buttonPanelAnswers = listOf(
        AnswerItem(R.string.quest_generic_hasFeature_no) { applyAnswer(NO) },
        AnswerItem(R.string.quest_generic_hasFeature_yes) { applyAnswer(YES) }
    )

    override val otherAnswers = listOf(
        AnswerItem(R.string.quest_bench_answer_picnic_table) { applyAnswer(PICNIC_TABLE) }
    )
}
